napari/utils/chunk/_request.py

```diff
@@ -0,0 +1,298 @@
+        self.timers[timer_name] = TimeSpan(start_seconds, end_seconds, args)
+
+    def load_chunks(self):
+        """Load all of our chunks now in this thread."""
+        self.process_id = os.getpid()
+        self.thread_id = threading.get_ident()
+
+        # Time the loading of all the chunks.
+        with self.time_block("load_chunks"):
+
+            # Simulate latency.
+            if self.load_seconds > 0:
+                time.sleep(self.load_seconds)
+
+            # Time the load of every chunk.
+            for key, array in self.chunks.items():
+                with self.time_block(key) as args:
+                    loaded_array = np.asarray(array)
+                    self.chunks[key] = loaded_array
+
+                    # Arguments for this PerfEvent.
+                    args['shape'] = loaded_array.shape
+                    args['nbytes'] = loaded_array.nbytes
+
+    def add_perf_events(self):
+        """Add perf events for this request.
+
+        This should be called in the main process after the chunks were
+        loaded in the worker. All this effort is not necessary with
+        threads, but we do it the same way for processes or threads
+        since it will work fine with either.
+        """
+        if not timers:
+            return  # we're not using perfmon
+
+        # Convert from time.time() to perf_counter_ns()
+        delta_ns = perf_counter_ns() - (time.time() * 1e9)
+
+        # Add a PerfEvent for each of our timers.
+        for name, time_span in self.timers.items():
+
+            # From seconds to nanoseconds
+            start_ns = time_span.start_seconds * 1e9 + delta_ns
+            end_ns = time_span.end_seconds * 1e9 + delta_ns
+
+            # Add the PerfEvent
+            timers.add_event(
+                PerfEvent(
+                    name,
+                    start_ns,
+                    end_ns,
+                    process_id=self.process_id,
+                    thread_id=self.thread_id,
+                    **time_span.args,
+                )
+            )
+
+    def transpose_chunks(self, order):
+        """Transpose all our chunks.
+
+        Parameters
+        ----------
+        order
+            Transpose the chunks with this order.
+        """
+        for key, array in self.chunks.items():
+            self.chunks[key] = array.transpose(order)
+
+    @property
+    def image(self):
+        """The image chunk if we have one or None.
+        """
+        return self.chunks.get('image')
+
+    @property
+    def thumbnail_source(self):
+        """The chunk to use as the thumbnail_source or None.
+        """
+        try:
+            return self.chunks['thumbnail_source']
+        except KeyError:
+            # For single-scale we use the image as the thumbnail_source.
+            return self.chunks.get('image')
+
+    def is_compatible(self, layer) -> bool:
+        """Return True if the given data is compatible with this request.
```

Comment:
```suggestion
        """Return True if layer.data id matches the data id of our ChunkKey."""
```

Reply:
The current way is better. The doc string describes the externally visible behavior of the function. As it says it determines "if the data is compatible with the request". Current code the code compares id's but this might change. There's no need for the doc string to describe the implementation details of the code here, since they might change over time. In general I think the doc string should describe _what_ a function does, but should not detail _how_ it is done, that's what to the code or internal comments are for.